01.md

```diff
@@ -127,13 +127,13 @@ Clients can send 3 types of messages, which must be JSON arrays, according to th
   "#<single-letter (a-zA-Z)>": <a list of tag values, for #e — a list of event ids, for #p — a list of pubkeys, etc.>,
   "since": <an integer unix timestamp in seconds, events must be newer than this to pass>,
   "until": <an integer unix timestamp in seconds, events must be older than this to pass>,
-  "limit": <maximum number of events relays SHOULD return in the initial query>
+  "limit": <maximum number of matched events relays SHOULD return in the initial query>
 }
 ```
 
-Upon receiving a `REQ` message, the relay SHOULD query its internal database and return events that match the filter, then store that filter and send again all future events it receives to that same websocket until the websocket is closed. The `CLOSE` event is received with the same `<subscription_id>` or a new `REQ` is sent using the same `<subscription_id>`, in which case relay MUST overwrite the previous subscription.
+Upon receiving a `REQ` message, the relay SHOULD query its internal database and return events that match the filters, then store those filters and send again all future events it receives to that same websocket until the websocket is closed. The `CLOSE` event is received with the same `<subscription_id>` or a new `REQ` is sent using the same `<subscription_id>`, in which case relay MUST overwrite the previous subscription.
 
-Filter attributes containing lists (`ids`, `authors`, `kinds` and tag filters like `#e`) are JSON arrays with one or more values. At least one of the arrays' values must match the relevant field in an event for the condition to be considered a match. For scalar event attributes such as `authors` and `kind`, the attribute from the event must be contained in the filter list. In the case of tag attributes such as `#e`, for which an event may have multiple values, the event and filter condition values must have at least one item in common.
+Filter attributes containing lists (`ids`, `authors`, `kinds` and tag conditions like `#e`) are JSON arrays with one or more values. At least one of the arrays' values must match the relevant field in an event for the condition to be considered a match. Scalar event attributes such as `pubkey` and `kind` must match the corresponding filter condition exactly. In the case of tag attributes such as `#e`, for which an event may have multiple values, the event and filter condition values must have at least one item in common.
```

Comment:
`authors` was wrong here. Replaced with `pubkey`.

Reply:
This sounds more confusing than before.

If there is a filter `{"kinds": [1,2,3]}` how does an event `{"kind": 1}` match it exactly?

Reply:
I see your point. I deleted the sentence and included the information in the filter format example.